### ISSUE TITLE
AdadeltaOptimizer numeric issue

### ISSUE THREAD

**Issue by darknights** [tensorflow/tensorflow#9058]
Hi,

I tested it with following script, and expected the model update to be -sqrt(0+1)/sqrt(1+1)*1 = -0.7071. However, the output is -0.866. Input gradient is 1, and confirmed with GradientDescentOptimizer. I also verified in rho=1/epsilon=1 the output is -1 as expected, and rho=1/epsilon=0 the output is nan as expected. Reading the implementation of adadelta didn't tell me why the output is -0.866 for rho=0/epsilon=1. What am I missing?

    def test_tf():
        data  = [0.5]
        label = [0]
        dim   = 1
        batch_size = 1
        import tensorflow as tf
        tf.reset_default_graph()

        x = tf.placeholder(tf.float32, [batch_size, dim])
        l = tf.placeholder(tf.int32, [batch_size, dim])
        W = tf.Variable(tf.zeros((dim,)), name='W')
        logits = x + W
        print(logits)
        loss = tf.losses.mean_squared_error(logits, l)
        #optimizer = tf.train.GradientDescentOptimizer(1)
        optimizer = tf.train.AdadeltaOptimizer(1, rho=0, epsilon=1)
        train = optimizer.minimize(loss)

        with tf.Session() as sess:
            sess.run(tf.global_variables_initializer())
            input_map = {x:[data], l:[label]}
            print('loss', loss.eval(input_map))
            sess.run(train, input_map)
            print('\n'.join([' {}\n'.format(p.eval()) for p in tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES)]))
        sess.close()

    test_tf()

**Comment by asimshankar** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by darknights** [comment]
I do think this is a bug though. Tried to generate the ground truth according to https://github.com/tensorflow/tensorflow/blob/a0d784bdd31b27e013a7eac58a86ba62e86db299/tensorflow/python/training/adadelta_test.py#L98

    def ground_truth():    
        accum = 0
        accum_update = 0
        grad = 1
        rho = 0
        epsilon = 1
        lr = 1
        tot_update = 0
        
        accum = accum * rho + (grad**2) * (1 - rho)
        update = (np.sqrt(accum_update + epsilon) *
                 (1. / np.sqrt(accum + epsilon)) * grad)
        accum_update = (accum_update * rho + (update**2) *
                       (1.0 - rho))
        tot_update += update * lr
        print('expected:', tot_update)

And the output is 0.7071 instead of 0.866
